### ISSUE TITLE
Support lazy task configuration for Gradle plugin

### ISSUE THREAD

**Issue by tasomaniac** [apollographql/apollo-kotlin#1454]
The Gradle plugin does not use the new lazy task configuration. This means that its tasks are eagerly created + it causes other dependant Android/Java/Kotlin tasks to be eagerly realized.

Integrate Lazy task configuration for build speed improvements. 

More info: https://docs.gradle.org/current/userguide/task_configuration_avoidance.html

**Comment by martinbonnin** [comment]
It's not possible at the moment because `Variant.registerJavaGeneratingTask` does not allow for lazy configuration. There's an issue in the google bug tracker: https://issuetracker.google.com/issues/117343589

**Comment by tasomaniac** [comment]
Yes I realized this shortly after trying to solve this. I forgot to come back and write it here. Thanks @martinbonnin for that. 

As far as I learned, Android team will replace this with a newer lazy FileProvider/Directory approach and will not directly work with Tasks. That's definitely better but that means that this is basically blocked.

Considering the schedule of Android Studio releases, this may take up to 1 year.
